include/alpaka/kernel/Traits.hpp

```diff
@@ -68,6 +68,19 @@ namespace alpaka
             }
         };
 
+        //! The trait for getting the warp size required by a kernel.
+        //!
+        //! \tparam TKernelFnObj The kernel function object.
+        //! \tparam TAcc The accelerator.
+        //!
+        //! The default implementation returns 0, which lets the accelerator compiler and runtime choose the warp size.
+        template<typename TKernelFnObj, typename TAcc, typename TSfinae = void>
+        struct WarpSize
+        {
+            static constexpr std::uint32_t warp_size = 0u;
+        };
```

Comment:
I would prefer this to be an integral constant:
```suggestion
        template<typename TKernelFnObj, typename TAcc, typename TSfinae = void>
        struct WarpSize : std::integral_constant<0>{};
```
Also please provide a shorthand:
```c++
template<typename TKernelFnObj, typename TAcc>
inline constexpr std::uint32_t warpSize = WarpSize<TKernelFnObj, TAcc>::value;
```